config/template.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const Joi = require('joi');
+const Job = require('./job');
+const Regex = require('./regex');
+
+const TEMPLATE_NAME = Joi
+            .string()
+            .regex(Regex.TEMPLATE_NAME)
+            .description('Name of the Template')
+            .example('node/npm-install');
+
+const TEMPLATE_VERSION = Joi
+            .string()
+            .regex(Regex.VERSION)
+            .description('Version of the Template')
+            .example('1.2.3');
+
+const TEMPLATE_DESCRIPTION = Joi
+            .string()
+            .description('Description of the Template')
+            .example('Installs npm modules');
+
+const TEMPLATE_MAINTAINER = Joi
+            .string()
+            .email()
+            .description('Maintainer of the Template')
+            .example('foo@bar.com');
+
+const TEMPLATE_CONFIG = Joi
+            .object()
+            .keys({
+                steps: Job.steps
```

Comment:
What about other fields?
In fact, since we decided it's going to be similar to `job`, we can reuse this: https://github.com/screwdriver-cd/data-schema/blob/master/config/job.js#L87